### PR TITLE
Add the DbgGetSymbolInfoAt API

### PR DESCRIPTION
**Pull request by m417z**
Mainly to overcome the restrictive `MAX_LABEL_SIZE` limit, and to be a able to get heavily templated symbols.

### CODE REVIEW

File: src/bridge/bridgemain.h [hunk 1]
```diff
@@ -334,6 +334,7 @@ typedef enum
     DBG_ANALYZE_FUNCTION,           // param1=BridgeCFGraphList* graph,  param2=duint entry
     DBG_MENU_PREPARE,               // param1=int hMenu,                 param2=unused
     DBG_GET_SYMBOL_INFO,            // param1=void* symbol,              param2=SYMBOLINFO* info
+    DBG_GET_SYMBOL_INFO_AT,         // param1=duint addr,                param2=SYMBOLINFO* info
```
Comment (mrexodia) [inline src/bridge/bridgemain.h:337]: New values can only be appended to the end, otherwise this would break things...

File: src/dbg/_exports.cpp [hunk 2]
```diff
@@ -1653,6 +1653,21 @@ extern "C" DLL_EXPORT duint _dbg_sendmessage(DBGMSG type, void* param1, void* pa
     }
     break;
 
+    case DBG_GET_SYMBOL_INFO_AT:
+    {
+        auto modbase = ModBaseFromAddr((duint)param1);
```
Comment (mrexodia) [inline src/dbg/_exports.cpp:1658]: This check is already done in `SymbolFromAddressExact`, so it can be removed.

**Review by mrexodia — CHANGES_REQUESTED** [review]

File: src/dbg/_exports.cpp (hunk 2 above)
Comment (m417z) [reply src/dbg/_exports.cpp:1658]: But the query is done again, I thought it wouldn't hurt. `ModInfoFromAddr` says it DOES _NOT_ USE LOCKS.

```cpp
MODINFO* ModInfoFromAddr(duint Address)
{
    //
    // NOTE: THIS DOES _NOT_ USE LOCKS
    //
    auto found = modinfo.find(Range(Address, Address));
    ///...
```

**Comment by mrexodia** [comment]
Thanks! I've been looking at this for a use case I have as well. I added the `startRva` and `endRva` parameters to `enumSymbols`, but then I saw the terrible API and gave up from there...

File: src/dbg/_exports.cpp (hunk 2 above)
Comment (mrexodia) [reply src/dbg/_exports.cpp]: The extra check doesn’t hurt, but it’s pointless and a (harmless) race condition.

The ModInfoFromAddr can only be used while holding a lock on the modules (hence the comment). This is done with the SHARED_ACQUIRE(?) macro and this ensures the module won’t be unloaded while symbols are being queried.

Comment (m417z) [reply src/dbg/_exports.cpp]: 👍 Removed the check.

Comment (mrexodia) [reply src/dbg/_exports.cpp]: Hm I just saw the issue, but I guess this will work fine.

**Comment by mrexodia** [comment]
Thanks!

**Pull request merged**